logstash-core/lib/logstash/agent.rb

```diff
@@ -3,20 +3,45 @@
 require "logstash/errors"
 require "logstash/config/cpu_core_strategy"
 require "logstash/pipeline"
+require "logstash/config/loader"
 require "uri"
 require "stud/trap"
 
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
+  attr_reader :logger, :runner
 
-  attr_writer :logger
-
-  def initialize
+  def initialize(logger, runner)
```

Comment:
I'd expect the agent to be given commands by the runner and not receive it as an argument.

having the agent be smart and interact with the runner makes it less flexible, the "add_pipelines" api has all this hidden meaning of getting the config from the runner.

regardless of the agent implementation, as long as it has the add_pipeline and execute api, if the cli args are `-e ""`,  then the runner should at least add a new pipeline and start the agent